golang.org/x/sys/unix.Timespec cannot generally be converted to syscall.Timespec

On armv7 FreeBSD, build of this module fails due to an inability to convert from `golang.org/x/sys/unix.Timespec` to `syscall.Timespec`:

    # github.com/tonistiigi/fsutil/copy
    vendor/github.com/tonistiigi/fsutil/copy/copy_unix.go:65:44: cannot convert StatAtime(st) (value of struct type syscall.Timespec) to type "golang.org/x/sys/unix".Timespec
    vendor/github.com/tonistiigi/fsutil/copy/copy_unix.go:65:74: cannot convert StatMtime(st) (value of struct type syscall.Timespec) to type "golang.org/x/sys/unix".Timespec

This is because `syscall.Timespec` has extra padding:

```
$ GOARCH=arm go doc golang.org/x/sys/unix Timespec
package unix // import "golang.org/x/sys/unix"

type Timespec struct {
	Sec  int64
	Nsec int32
	// Has unexported fields.
}

func NsecToTimespec(nsec int64) Timespec
func (ts *Timespec) Nano() int64
func (ts *Timespec) Unix() (sec int64, nsec int64)
$ GOARCH=arm go doc syscall Timespec              
package syscall // import "syscall"

type Timespec struct {
	Sec       int64
	Nsec      int32
	Pad_cgo_0 [4]byte
}

func NsecToTimespec(nsec int64) Timespec
func (ts *Timespec) Nano() int64
func (ts *Timespec) Unix() (sec int64, nsec int64)
```

We apply the following patch on FreeBSD to avoid this issue; please consider taking it:

```diff
--- copy/copy_unix.go.orig	2026-07-06 12:19:07 UTC
+++ copy/copy_unix.go
@@ -62,7 +62,9 @@ func (c *copier) copyFileTimestamp(fi os.FileInfo, nam
 	}
 
 	st := fi.Sys().(*syscall.Stat_t)
-	timespec := []unix.Timespec{unix.Timespec(StatAtime(st)), unix.Timespec(StatMtime(st))}
+	atime := unix.Timespec{Sec: StatAtime(st).Sec, Nsec: StatAtime(st).Nsec}
+	mtime := unix.Timespec{Sec: StatMtime(st).Sec, Nsec: StatMtime(st).Nsec}
+	timespec := []unix.Timespec{atime, mtime}
 	if err := unix.UtimesNanoAt(unix.AT_FDCWD, name, timespec, unix.AT_SYMLINK_NOFOLLOW); err != nil {
 		return errors.Wrapf(err, "failed to utime %s", name)
 	}
```

--- a/copy/copy_linux.go
+++ b/copy/copy_linux.go
@@ -62,7 +62,11 @@ func (c *copier) copyFileTimestamp(fi os.FileInfo, name string) error {
 	}
 
 	st := fi.Sys().(*syscall.Stat_t)
-	timespec := []unix.Timespec{unix.Timespec(StatAtime(st)), unix.Timespec(StatMtime(st))}
+	atime, mtime := StatAtime(st), StatMtime(st)
+	timespec := []unix.Timespec{
+		{Sec: atime.Sec, Nsec: atime.Nsec},
+		{Sec: mtime.Sec, Nsec: mtime.Nsec},
+	}
 	if err := unix.UtimesNanoAt(unix.AT_FDCWD, name, timespec, unix.AT_SYMLINK_NOFOLLOW); err != nil {
 		return errors.Wrapf(err, "failed to utime %s", name)
 	}

--- a/copy/copy_unix.go
+++ b/copy/copy_unix.go
@@ -62,7 +62,11 @@ func (c *copier) copyFileTimestamp(fi os.FileInfo, name string) error {
 	}
 
 	st := fi.Sys().(*syscall.Stat_t)
-	timespec := []unix.Timespec{unix.Timespec(StatAtime(st)), unix.Timespec(StatMtime(st))}
+	atime, mtime := StatAtime(st), StatMtime(st)
+	timespec := []unix.Timespec{
+		{Sec: atime.Sec, Nsec: atime.Nsec},
+		{Sec: mtime.Sec, Nsec: mtime.Nsec},
+	}
 	if err := unix.UtimesNanoAt(unix.AT_FDCWD, name, timespec, unix.AT_SYMLINK_NOFOLLOW); err != nil {
 		return errors.Wrapf(err, "failed to utime %s", name)
 	}

